validation/longhorn/longhorn_test.go

```diff
@@ -0,0 +1,314 @@
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/rancher/tests/actions/charts"
+	"github.com/rancher/tests/actions/cloudprovider"
+	namespaceActions "github.com/rancher/tests/actions/namespaces"
+	"github.com/rancher/tests/actions/projects"
+	"github.com/rancher/tests/actions/rbac"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	appv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	longhornNamespace                 = "longhorn-system"
+	longhornChartName                 = "longhorn"
+	longhornProjectName               = "longhorn-test"
+	longhornStorageClass              = "longhorn"
+	longhornStaticDefaultStorageClass = "longhorn-static"
+	debugNamespace                    = namegenerator.AppendRandomString("debug-")
+)
+
+type LonghornTestSuite struct {
+	suite.Suite
+	client            *rancher.Client
+	session           *session.Session
+	cluster           *clusters.ClusterMeta
+	project           *management.Project
+	payloadOpts       charts.PayloadOpts
+	installedLonghorn bool
+}
+
+func (l *LonghornTestSuite) TearDownSuite() {
+	l.session.Cleanup()
+}
+
+func (l *LonghornTestSuite) SetupSuite() {
+	l.session = session.NewSession()
+
+	client, err := rancher.NewClient("", l.session)
+	require.NoError(l.T(), err)
+
+	l.client = client
+
+	l.cluster, err = clusters.NewClusterMeta(client, client.RancherConfig.ClusterName)
+	require.NoError(l.T(), err)
+
+	projectConfig := &management.Project{
+		ClusterID: l.cluster.ID,
+		Name:      longhornProjectName,
+	}
+
+	l.project, err = client.Management.Project.Create(projectConfig)
+	require.NoError(l.T(), err)
+
+	// Get latest versions of longhorn
+	latestLonghornVersion, err := l.client.Catalog.GetLatestChartVersion(longhornChartName, catalog.RancherChartRepo)
+	require.NoError(l.T(), err)
+
+	l.payloadOpts = charts.PayloadOpts{
+		Namespace: longhornNamespace,
+		Host:      l.client.RancherConfig.Host,
+		InstallOptions: charts.InstallOptions{
+			Cluster:   l.cluster,
+			Version:   latestLonghornVersion,
+			ProjectID: l.project.ID,
+		},
+	}
+}
+
+func (l *LonghornTestSuite) TestChartInstall() {
+	chart, err := shepherdCharts.GetChartStatus(l.client, l.cluster.ID, longhornNamespace, longhornChartName)
+	require.NoError(l.T(), err)
+
+	if chart.IsAlreadyInstalled {
+		l.T().Skip("Skipping installation test because Longhorn is already installed")
+	}
+
+	l.T().Logf("Installing Longhorn chart in cluster [%v] with latest version [%v] in project [%v] and namespace [%v]", l.cluster.Name, l.payloadOpts.Version, l.project.Name, l.payloadOpts.Namespace)
+	err = charts.InstallLonghornChart(l.client, l.payloadOpts, nil)
+	require.NoError(l.T(), err)
+	l.installedLonghorn = true
+}
+
+func (l *LonghornTestSuite) TestVolumeCreationOnWorkload() {
```

Comment:
Typically with charts, we write the test for a given chart as follows:

* install the chart
* verify the chart is working correctly

in your tests, you split these two into different tests. Can you instead combine TestChartInstall and TestVolumeCreatonOnWorkoad into one, probably just TestChart or something similar?